### PR TITLE
GH-552: Remove references to boot-clj and its files.

### PR DESCRIPTION
**Pull request by tbl3rd**
### Purpose

- https://broadinstitute.atlassian.net/browse/GH-552

### Changes

- Looks like this relies on Clojure CLI features not supported by the version used in checks.

### Review Instructions

- No instructions.

### CODE REVIEW

**Comment by jack-r-warren** [comment]
Got a funky error. I ran

```
╰─ git clone https://github.com/broadinstitute/wfl.git
Cloning into 'wfl'...
remote: Enumerating objects: 457, done.
remote: Counting objects: 100% (457/457), done.
remote: Compressing objects: 100% (193/193), done.
remote: Total 4506 (delta 302), reused 350 (delta 236), pack-reused 4049
Receiving objects: 100% (4506/4506), 3.46 MiB | 3.92 MiB/s, done.
Resolving deltas: 100% (2633/2633), done.


╰─ cd wfl

╰─ git checkout tbl/GH-552-boot-less
Branch 'tbl/GH-552-boot-less' set up to track remote branch 'tbl/GH-552-boot-less' from 'origin'.
Switched to a new branch 'tbl/GH-552-boot-less'

╰─ make TARGET=prebuild
git clone git@github.com:broadinstitute/pipeline-config.git /Users/jwarren/wfl/derived/2p/pipeline-config
Cloning into '/Users/jwarren/wfl/derived/2p/pipeline-config'...
remote: Enumerating objects: 38, done.
remote: Counting objects: 100% (38/38), done.
remote: Compressing objects: 100% (26/26), done.
remote: Total 585 (delta 22), reused 11 (delta 9), pack-reused 547
Receiving objects: 100% (585/585), 112.31 KiB | 3.04 MiB/s, done.
Resolving deltas: 100% (302/302), done.
git clone git@github.com:broadinstitute/warp.git /Users/jwarren/wfl/derived/2p/warp
Cloning into '/Users/jwarren/wfl/derived/2p/warp'...
remote: Enumerating objects: 310, done.
remote: Counting objects: 100% (310/310), done.
remote: Compressing objects: 100% (163/163), done.
remote: Total 5988 (delta 202), reused 198 (delta 136), pack-reused 5678
Receiving objects: 100% (5988/5988), 36.91 MiB | 10.04 MiB/s, done.
Resolving deltas: 100% (3693/3693), done.
clojure -X wfl.boot/prebuild
Execution error (FileNotFoundException) at java.io.FileInputStream/open0 (FileInputStream.java:-2).
-X (No such file or directory)

Full report at:
/var/folders/yt/3cpzfkys3h939n50fphnhhrc0000gp/T/clojure-14881614051169622468.edn
make[1]: *** [/Users/jwarren/wfl/derived/api/prebuild.timestamp] Error 1
make: *** [api] Error 2
```

Looking into it myself. `openjdk 14.0.1 2020-04-14`

**Comment by tbl3rd** [comment]
That's the same error the PR checks reproted this morning.
Let me know if updating the clojure CLI does not fix it.

**Comment by jack-r-warren** [comment]
> That's the same error the PR checks reproted this morning.
> Let me know if updating the clojure CLI does not fix it.

Yup, that did it, thank you!

**Review by jack-r-warren — APPROVED** [review]
Looks good to me. I built using this and pushed it to my sandbox and things worked as expected (my sandbox reports a version of 0.5.0 because that's what the version.edn had but I did *not* release that, the images and everything else has a version of `0.5.0-jwarren-no-boot-1`)

**Comment by tbl3rd** [comment]
Cool. Thanks.

File: api/module.mk [hunk 1]
```diff
@@ -3,20 +3,23 @@
...
 DERIVED_TARGET_DIR    := $(DERIVED_MODULE_DIR)/target
-BOOT_PROJECT          := $(MODULE_DIR)/build.boot
 CLOJURE_PROJECT       := $(MODULE_DIR)/deps.edn
-LEIN_PROJECT          := $(MODULE_DIR)/project.clj
 
-CLEAN_DIRS  += $(CPCACHE_DIR)
-CLEAN_FILES += $(LEIN_PROJECT) $(MODULE_DIR)/wfl
+API_DIR      := $(CLASSES_DIR)/wfl/api
+POM_IN       := $(MODULE_DIR)/pom.xml
+POM_OUT      := $(SRC_DIR)/META-INF/maven/org.broadinstitute/wfl/pom.xml
```
Comment (ehigham) [inline api/module.mk:19]: If these are generated, is it possible to output them into `dervied/api`?

File: docs/md/terra.md [hunk 2]
```diff
@@ -172,21 +172,17 @@ WFL is designed around several novel concepts.
...
-    to bootstrap WFL from source. And `boot-clj` is a single file:
-    its own installer.
-
-    Similarly, `boot build` builds WFL, and `boot deploy` deploys
-    it to GAE.
-
-    WFL attempts to be self-describing and self-documenting. It
-    includes monitoring and diagnostic modules for tracking workload
-    progress and debugging failures.
+    `dsde-pipelines` repository.
```
Comment (samanehsan) [inline docs/md/terra.md:181]: This is now `warp`

File: api/module.mk [hunk 3]
```diff
@@ -28,22 +31,31 @@ TEST_SCM_SRC       = $(shell $(FIND) $(TEST_DIR) -type f -name "*.$(CLJ)")
...
-$(PREBUILD): $(MODULE_DIR)/wfl
+$(PREBUILD): $(SCM_SRC) $(SCM_RESOURCES)
 	@$(MKDIR) $(DERIVED_RESOURCES_DIR) $(DERIVED_SRC_DIR)
-	$(BOOT) prebuild
+	$(CLOJURE) -X wfl.boot/prebuild
 	@$(TOUCH) $@
 
-$(MODULE_DIR)/wfl: $(BOOT_PROJECT)
-	$(LN) $(BOOT_PROJECT) $@
+$(POM_IN): $(PREBUILD) $(CLOJURE_PROJECT)
```
Comment (ehigham) [inline api/module.mk:39]: Not sure if you need to depend on PREBUILD since you already have BUILD -> POM_OUT -> POM_IN? and BUILD needs PREBUILD?

**Review by ehigham — APPROVED** [review]

File: api/src/wfl/boot.clj [hunk 4]
```diff
@@ -1,7 +1,8 @@
 (ns wfl.boot
```
Comment (samanehsan) [inline api/src/wfl/boot.clj:1]: Could we call this `wfl.build` instead?

**Review by samanehsan — APPROVED** [review]

File: api/module.mk (hunk 1 above)
Comment (tbl3rd) [reply api/module.mk:19]: Evidently, not yet -- at least, not POM_IN from the command line.

https://github.com/clojure/tools.deps.alpha/blob/c971a662730e75ee05186a20a776bb37390ee127/src/main/clojure/clojure/tools/cli/api.clj#L85

Looking into POM_OUT now, and it should be possible to load deps as a library.

File: api/src/wfl/boot.clj (hunk 4 above)
Comment (tbl3rd) [reply api/src/wfl/boot.clj:1]: I'll do that in a follow-up to maintain blame.

Comment (samanehsan) [reply api/src/wfl/boot.clj:1]: Ooh good call!

File: api/module.mk (hunk 3 above)
Comment (tbl3rd) [reply api/module.mk:39]: The pom needs the calculated version -- at least in principle.

Comment (tbl3rd) [reply api/module.mk:39]: Actually, I guess only POM_OUT needs the version, since POM_IN ignores it.

Comment (tbl3rd) [reply api/module.mk:39]: But now POM_OUT has more than one direct dependency. :sigh:

File: docs/md/terra.md (hunk 2 above)
Comment (tbl3rd) [reply docs/md/terra.md]: Yep. Thanks. I should have caught this when purging dsde-pipelines.

**Pull request merged**